### PR TITLE
Downlaod song method

### LINKED ISSUE #10: Pull songs from youtube
Collect songs from youtube and download them to disk

### PR DESCRIPTION
**Pull request by julySteuer**

### CODE REVIEW

File: src/requirements.txt
Comment (SFSeeger) [inline src/requirements.txt:1]: Please remove duplicate file

File: src/uploader/views/submission_views.py [hunk 1]
```diff
@@ -18,6 +19,7 @@ class SubmissionCreateView(ConfiguredLoginViewMixin, CreateView):
     template_name = "uploader/upload_form.html"
     success_url = reverse_lazy("index")
     update_name = "update-from-youtube"
+    submit_action = download_song
```
Comment (SFSeeger) [inline src/uploader/views/submission_views.py:22]: is this tested?

File: src/uploader/views/submission_views.py [hunk 2]
```diff
@@ -36,3 +41,13 @@ class SubmissionUpdateView(ConfiguredLoginViewMixin, UpdateView):
...
     success_url = reverse_lazy("index")
+    submit_action = download_song
+
+    def form_valid(self, form):
+        submission = Submission.objects.get(user=self.request.user)
+        submission.song_url = form.cleaned_data["song_url"]
+        submission.start_time = form.cleaned_data["start_time"]
+        submission.end_time = form.cleaned_data["end_time"]
+        submission.save()
+        async_task(self.submit_action, submission.song_url, submission)
```
Comment (SFSeeger) [inline src/uploader/views/submission_views.py]: Too many arguments => Runtime error

File: src/uploader/views/submission_views.py [hunk 3]
```diff
@@ -27,7 +29,10 @@ def get(self, request: HttpRequest, *args: str, **kwargs: Any) -> HttpResponse:
     def form_valid(self, form):
         if Submission.objects.filter(user=self.request.user).exists():
             return HttpResponse(status=412)
-        Submission.objects.create(**form.cleaned_data, user=self.request.user)
+        submission = Submission.objects.create(
+            **form.cleaned_data, user=self.request.user
+        )
+        async_task(self.submit_action, submission.song_url, submission)
```
Comment (SFSeeger) [inline src/uploader/views/submission_views.py]: Too many arguments => Runtime error

**Review by SFSeeger — COMMENTED** [review]
Please fix comments
Add `MEDIA_ROOT="/var/www/data/media/"` to `settings/prod.py`
And `MEDIA_URL="/media/"` to` settings/base.py`

File: src/uploader/views/submission_views.py (hunk 1 above)
Comment (julySteuer) [reply src/uploader/views/submission_views.py:22]: grafik
I dont know this existed

Comment (SFSeeger) [reply src/uploader/views/submission_views.py:22]: ah well, there is a django thing simmilar to this, but this is just a var to reuse the view

File: src/requirements.txt
Comment (SFSeeger) [inline src/requirements.txt:1]: pls delete file

File: src/uploader/views/submission_views.py [hunk 4]
```diff
@@ -36,3 +41,13 @@ class SubmissionUpdateView(ConfiguredLoginViewMixin, UpdateView):
...
     success_url = reverse_lazy("index")
+    submit_action = download_song
+
+    def form_valid(self, form):
+        submission = Submission.objects.get(user=self.request.user)
+        submission.song_url = form.cleaned_data["song_url"]
+        submission.start_time = form.cleaned_data["start_time"]
+        submission.end_time = form.cleaned_data["end_time"]
+        submission.save()
+        async_task(self.submit_action, submission)
```
Comment (SFSeeger) [inline src/uploader/views/submission_views.py:52]: remove unnecessary argument

File: src/uploader/views/submission_views.py [hunk 5]
```diff
@@ -27,7 +29,10 @@ def get(self, request: HttpRequest, *args: str, **kwargs: Any) -> HttpResponse:
     def form_valid(self, form):
         if Submission.objects.filter(user=self.request.user).exists():
             return HttpResponse(status=412)
-        Submission.objects.create(**form.cleaned_data, user=self.request.user)
+        submission = Submission.objects.create(
+            **form.cleaned_data, user=self.request.user
+        )
+        async_task(self.submit_action, submission)
```
Comment (SFSeeger) [inline src/uploader/views/submission_views.py:35]: pls remove unnecessary argument

**Review by SFSeeger — APPROVED** [review]

**Pull request merged**